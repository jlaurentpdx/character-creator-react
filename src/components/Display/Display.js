export default function Display({ selection, feature }) {
  return (
    <div
      className={`bg ${feature}`}
      role="img"
      style={{
        backgroundImage: `url(${
          process.env.PUBLIC_URL + `/char_images/${selection}-${feature}.png`
        })`,
      }}
    ></div>
  );
}
