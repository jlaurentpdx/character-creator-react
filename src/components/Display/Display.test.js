import { render, screen } from '@testing-library/react';
import Display from './Display';

test('Display should render a div with image information', () => {
  const { container } = render(<Display selection="bird" feature="head" />);
  const image = screen.getByRole('img');
  expect(image.style.backgroundImage).toContain('/char_images/bird-head.png');

  expect(container).toMatchSnapshot();
});
