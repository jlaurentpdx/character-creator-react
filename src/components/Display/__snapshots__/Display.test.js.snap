// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display should render a div with image information 1`] = `
<div>
  <div
    class="bg head"
    role="img"
    style="background-image: url(/char_images/bird-head.png);"
  />
</div>
`;
